scripts/tools/zap/generate.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python3
+#
+#    Copyright (c) 2021 Project CHIP Authors
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import json
+import os
+from pathlib import Path
+import sys
+
+# Check python version
+if sys.version_info[0] < 3:
+    print('Must use Python 3. Current version is ' + str(sys.version_info[0]))
+    exit(1)
+
+argc = len(sys.argv) - 1
```

Comment:
could you use the argparse module instead? provides nice --help and more extensible